training.qmd

```diff
@@ -1,95 +1,663 @@
-- Data Privacy
-- Ethical Considerations
 
-## Conclusion
+ as orthogonal matrices. This helps in preserving the gradients during backpropagation and can be particularly useful in recurrent neural networks (RNNs).
+ Uniform and Normal Initialization:
+
+Weights are initialized with random values drawn from a uniform or normal distribution. The choice between uniform and normal depends on the specific requirements of the model and the activation functions used.
+Choosing the right initialization method depends on the architecture of the neural network, the activation functions, and the specific problem being solved. Experimentation with different techniques is often necessary to find the most suitable initialization for a given scenario.
+
+* Xavier/Glorot Initialization
+
+    Proposed by Xavier Glorot and Yoshua Bengio, this initialization is designed to work well with activation functions like tanh or logistic sigmoid. It initializes weights with random values drawn from a distribution with mean 0 and variance 2 / (number of input units + number of output units).
+
+* He Initialization
+
+    Proposed by Kaiming He et al., this initialization is tailored for ReLU (Rectified Linear Unit) activation functions. It initializes weights with random values drawn from a distribution with mean 0 and variance 2 / number of input units. This helps to mitigate the vanishing gradient problem often associated with deep networks.
+
+## Activation Functions
+
+Activation functions play a critical role in neural networks by introducing non-linearities into the model. These non-linearities enable neural networks to learn complex relationships and patterns in data, making them capable of solving a wide range of problems. Here are some commonly used activation functions:
 
-Explanation: A summary helps to consolidate the key points of the chapter, aiding in better retention and understanding of the material.
+* Rectified Linear Unit (ReLU):
+
+    ReLU is a popular activation function that returns zero for
+    negative input values and passes positive input values
+    unchanged. It is computationally efficient and has been widely
+    used in deep learning models.
+
+* Sigmoid Function
+
+    The sigmoid function, also known as the logistic function,
+    squashes input values between 0 and 1. It is often used in the
+    output layer of binary classification models, where the goal is to
+    produce probabilities.
+
+* Hyperbolic Tangent Function (tanh):
+
+    The hyperbolic tangent function is similar to the sigmoid but
+    squashes input values between -1 and 1. It is often used in hidden
+    layers of neural networks, especially when zero-centered outputs
+    are desired.
+
+![Common activation functions](https://1394217531-files.gitbook.io/~/files/v0/b/gitbook-legacy-files/o/assets%2F-LvBP1svpACTB1R1x_U4%2F-LvNWUoWieQqaGmU_gl9%2F-LvO3qs2RImYjpBE8vln%2Factivation-functions3.jpg?alt=media&token=f96a3007-5888-43c3-a256-2dafadd5df7c){width=70%}
```

Comment:
Diagram shows Linear as well. Could be worth discussing that here as well about when it might be desired

Reply:
Good point, done - thanks!